### DISCUSSION TITLE
Move all *-examples repos (interactive-examples, dom-examples, css-examples, etc.) into the mdn/content repo

### DISCUSSION THREAD

**Discussion by sideshowbarker** [mdn/mdn-community#431] [Content]
We should move all the *-examples repos (interactive-examples, dom-examples, css-examples, etc.) into the mdn/content repo — along with all their commit histories and open issues.

The only thing that blocks us from being able to do that right now is the existing open PRs in each of those repos. I don’t believe we’d be able to move the PRs — they’d instead either need to be merged or closed first, or we’d need for the PR authors to re-open them against the mdn/content repo after the move.

**Rationale:** None of the maintainers with PR merge perms and issue-triage perms seem to be actively watching the issues and PRs in all the *-interactive repos, and not triaging them — as evidenced by some of the repos having unmerged (but mergeable) PRs that are more than a year old. And all those repos are just content — though specifically, code-example content rather than prose. And I don’t think any of the maintainers would actually prefer needing to watch and triage 20+ separate content repos, when they could instead just be watching one repo, with all the content in one place.

**Comment by hamishwillee** [comment]
This is probably a good idea.

One queue to manage will be helpful for those people who manage those repos. I don't, so it will probably make my life a little harder in terms of larger inflow of github actions. 

We'll need to be careful to make it clear how to build each particular set of examples that are now embedded as part of the bigger docset, and fix any macros, docs, and so on that expect the sources to be in a particular location.

**Comment by yarusome** [comment]
A tangential question: Would this transition enable l10n of these examples? It seems that the structure of these examples might need redesigning to avoid duplicate contents so that l10n repos can store localizable strings only. (But I'd be equally happy to see examples localizable even if l10n repos have to copy duplicate contents.)

**Reply by SphinxKnight** [reply]
@yarusome this is quite a late feedback but I'd consider examples as "content as a whole". At the moment, the localized prose/MD content could also be considered as "duplicate contents" but it's localized nonetheless.

Also, the only solution in place if l10n teams want to fully localize embedded examples is to have forks in place. This doesn't scale well in terms of maintainability and security.

**Reply by DomOBU** [reply]
The textual content of the examples is generally of little importance; moreover, the textual content is simple and the sentences have a basic structure. So if the examples are in English, I don't think readers will mind.

What's more, a minimum of English is necessary in IT.

If, in exceptional cases, it should prove necessary to understand the content of the text, online translators are available.

Using English examples would simplify the translators' work and lighten their load.

**Comment by OnkarRuikar** [comment]
- https://github.com/mdn/content/issues/28174#issuecomment-1715266324
^ this is one more reason for doing this. Because it is a lot of work to replicate and maintain same tooling (linting and workflows) in every repo.

**Comment by estelle** [comment]
At the [#writing-docs meeting](https://docs.google.com/document/d/1AtzS1D6H4fsyHoSY52sJPHG57MBBqjGWievQV4kAuHY/edit) today, this PR was an agenda item. The general consensus was:
1.  All the examples should be in a single repository,
2. That repo should not necessarily be /content/, as /content/ is all markdown and the examples are code
3.  the maintainers of /content/ should have the same rights on that repo as they do on /content/
4. We should reach out the Mozilla as they are the ones who can make this happen. 

Not discussed at the meeting, but i think we are in agreement from other discussions, while the Github repo may be separate from /content/, the content should be served on developer.mozilla.org or mdn.github.io, but not in 15 different locations.
